src/components/Identity/identityUtil.js

```diff
@@ -0,0 +1,61 @@
+import { AUTH_STATES } from "./constants";
+import isObject from "../../utils/isObject";
+
+const validateCustomerIDs = customerIDs => {
+  if (!isObject(customerIDs)) {
+    throw new Error(
+      "Invalid customer ID format. Each namespace should be an object"
+    );
+  }
+  Object.keys(customerIDs).forEach(customerID => {
+    if (!isObject(customerIDs[customerID])) {
+      throw new Error(
+        "Invalid customer ID format. Each namespace should be an object"
+      );
+    }
+    if (!customerIDs[customerID].id) {
+      throw new Error(
+        "Invalid customer ID format. Each namespace object should have an id"
+      );
+    }
+  });
+};
+const normalizeCustomerIDs = customerIDs => {
+  const normalized = {};
+  Object.keys(customerIDs).forEach(customerID => {
+    const currentID = customerIDs[customerID];
+    normalized[customerID] = { ...currentID };
+    if (
+      currentID.authState !== AUTH_STATES.AUTHENTICATED ||
+      currentID.authState !== AUTH_STATES.LOGGED_OUT
+    ) {
+      normalized[customerID].authState = AUTH_STATES.UNKNOWN;
+    }
+  });
+  return normalized;
+};
+const serializeCustomerIDs = customerIDs => {
```

Comment:
Do we need this? Can we simply stringify the JSON instead, then hash that?

Reply:
Only advantage is size, Stringify adds a `"` character before and after every key and its value.

Reply:
If we're using SHA to hash the serialized customer IDs, it doesn't matter what size the serialized value is; the SHA hash output will always be the same size regardless. I agree with @jfkhoury that we should use `JSON.stringify()` here, but regardless of what we use, we need to be smarter about how we serialize. Here's why:

```
JSON.stringify({ a: 1, b: 2 }) === JSON.stringify({ b: 2, a: 1 });
```

Conceptually, both objects represent the same thing to the user, but the serialized value, and thus the hashes, will be different. We probably need to break the object down and alphabetize the entries into an array of entries sorted alphanumerically or something like that, such that the output from the two inputs will always be equal. An automated test should be added for this too.

Reply:
@Aaronius ,
I updated the normalizeCustomerIds method to sort according to id namespaces and then take only id and authState in the order.